paasta_tools/check_spark_jobs.py

```diff
@@ -0,0 +1,170 @@
+import argparse
+import datetime
+import logging
+from collections import defaultdict
+
+import requests
+
+from paasta_tools import mesos_tools
+from paasta_tools.monitoring_tools import send_event
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import list_services
+
+
+logger = logging.getLogger(__name__)
+
+
+JUPYTER_PREFIX = 'jupyterhub_'
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Reports long-running Spark frameworks.',
+    )
+    parser.add_argument(
+        '--min-hours',
+        type=float,
+        help='Report frameworks that have been registered for more than this duration',
+        default=0,
+    )
+    parser.add_argument(
+        '--no-notify',
+        action='store_true',
+        help='Skip notifying the teams that own each framework',
+    )
+    return parser.parse_args()
+
+
+def get_time_running(framework):
+    registered_time = datetime.datetime.fromtimestamp(framework['registered_time'])
+    return datetime.datetime.now() - registered_time
+
+
+def get_spark_properties(framework):
+    webui_url = framework.get('webui_url')
+    env_endpoint = f'{webui_url}/api/v1/applications/{framework.id}/environment'
+
+    try:
+        response = requests.get(env_endpoint, timeout=5)
+    except (requests.exceptions.Timeout, requests.exceptions.ConnectionError) as e:
+        logger.warning(f'Unable to connect to {env_endpoint}: {e!r}')
+        return None
+
+    if response.status_code != 200:
+        logger.warning(f'Bad response from {env_endpoint}: {response.status_code}')
+        return None
+
+    try:
+        return response.json()['sparkProperties']
+    except (ValueError, KeyError) as e:
+        logger.warning(f'Unable to get sparkProperties for {framework.id}: {e!r}')
```

Comment:
It might be helpful to print the response (not json) to debug to help figure out "why" we are getting a Error here.